tasks/configure-fpm.yml

```diff
@@ -69,9 +69,16 @@
   when: php_enable_php_fpm
   notify: restart php-fpm
 
+- name: systemd restart php-fpm
+  systemd:
+    name: "{{ php_fpm_daemon }}.service"
+    daemon_reload: yes
+    state: restarted
+  when: php_enable_php_fpm and ansible_service_mgr == 'systemd'
+
 - name: Ensure php-fpm is started and enabled at boot (if configured).
   service:
     name: "{{ php_fpm_daemon }}"
     state: started
     enabled: yes
-  when: php_enable_php_fpm
+  when: php_enable_php_fpm and ansible_service_mgr != 'systemd'
```

Comment:
Besides Debian, this should work correctly whether the service manager is systemd or initv.